chore: lint and doc cleanup

nyurik opened the pull request:
A few more minor cleanups, mostly in documentation

joshka commented:
I notice a bunch of checks are failing on this.

It looks like this change effectively ignores a bunch of places that could be better written. I wonder if it would be worth leaving the lints for those on rather than ignoring them, with a goal to fix the issues rather than ignore them.

nyurik commented:
> I notice a bunch of checks are failing on this.
> 
> It looks like this change effectively ignores a bunch of places that could be better written. I wonder if it would be worth leaving the lints for those on rather than ignoring them, with a goal to fix the issues rather than ignore them.

ok, makes sense, i removed the clippy changes for now

joshka commented on src/buffer.rs:
```diff
@@ -531,11 +531,9 @@ impl Debug for Buffer {
-    ///     { x: 1, y: 2, fg: Color::Red, bg: Color::Blue, modifier: Modifier::BOLD }
+    /// * `area`: displayed as `Rect { x: 1, y: 2, width: 3, height: 4 }`
+    /// * `content`: displayed as a list of strings representing the content of the buffer
+    /// * `styles`: displayed as a list of `{ x: 1, y: 2, fg: Color::Red, bg: Color::Blue, modifier: Modifier::BOLD }`
```
I'd suggest leaving this as a 100 char line limit (that limit is used quite a few places in the rust ecosystem). Objectively it helps to avoid horizontal scrolling or wrapping being necessary to read code:

E.g.:
<img width="1440" alt="image" src="https://github.com/tui-rs-revival/ratatui/assets/381361/ac6554a2-112f-45e1-aa57-ecae51221d3f">

vs current:

<img width="1440" alt="image" src="https://github.com/tui-rs-revival/ratatui/assets/381361/82a323c1-d252-42e8-8da9-b5e7ec221ad8">

mindoodoo commented on src/backend/mod.rs:
```diff
@@ -38,6 +38,7 @@ pub trait Backend {
     /// Insert `n` line breaks to the terminal screen
     fn append_lines(&mut self, n: u16) -> io::Result<()> {
         // to get around the unused warning
+        #[allow(clippy::no_effect_underscore_binding)]
         let _n = n;
         Ok(())
     }
```
```suggestion
    /// Insert `n` line breaks to the terminal screen
    #[allow(unused)]
    fn append_lines(&mut self, n: u16) -> io::Result<()> {
        Ok(())
    }
```

I don't understand why the `let _n = n` was used in the first place as this removes that warning, maybe wasn't an available feature before, or am I missing something ?

joshka replied:
My guess is that there is a concern that `n` in the doc comment and `_n` in the parameter list are noth the same, or that `_n` doesn't match the trait perhaps? Either way, I think probably just using _n in the parameter list is good enough.

mindoodoo commented:
Wasn't exactly sure for the conventional commit (PR title) as this is both linting and a couple doc fixes so marked it as unscoped chore, maintainers feel free to change it to something more adequate if you have any ideas.

joshka commented:
>Wasn't exactly sure for the conventional commit (PR title) as this is both linting and a couple doc fixes so marked it as unscoped chore, maintainers feel free to change it to something more adequate if you have any ideas.

chore is fine. I made the changes that we both suggested (just used _n instead of a lint exception)

nyurik commented:
@joshka were the Cargo.toml changes intentional?

nyurik commented:
Ah, I see - you moved the indoc into dependencies because of the format usage, plus sorted them.  Slightly outside of the initial scope, but sure, not a biggie :)

I removed the no-longer-needed comment about the underscore, and rebased on main
